go/cli_installation/README.md

```diff
@@ -0,0 +1,64 @@
+# CLI Installation
+
+This program demonstrates how to use the installation capabilities of the Automation API. By default Automation API expects the `pulumi` binary to be available on your `$PATH`. Using the installation capabilities you can customize which binary should be used and let the Automation API manage the installation.
+
+Running this program is just like any other Go program. No invocation through the Pulumi CLI required:
+
+```shell
+$ go run main.go
+Installing pulumi into /var/folders/4q/3skx3qb5191bjhkchcbbrmmm0000gn/T/cli_installation_exmple1292766419
+Created/Selected stack "dev"
+Starting update
+Updating (dev)
+
+View Live: https://app.pulumi.com/julienp/autoInstall/dev/updates/5
+
+
+ +  pulumi:pulumi:Stack autoInstall-dev creating (0s)
+@ Updating.....
+ +  random:index:RandomPet Fluffy creating (0s)
+ +  random:index:RandomPet Fluffy created (0.72s)
+@ Updating....
+ +  pulumi:pulumi:Stack autoInstall-dev created (3s)
+Outputs:
+    pet.id: "sweet-kangaroo"
+
+Resources:
+    + 2 created
+
+Duration: 4s
+
+Update succeeded!
+pet: capable-swine
```

Comment:
Why is this different than the output above?  Having a quick look at the code I thought they should be the same, what am I missing?

Reply:
Umm ... I ran the example and got `capable-swine` as pet, which I thought was a really great name. Then I updated the code a bit, destroyed the stack and ran the example again. This time I got a different name I didn't like as much, so I edited the last line in the output to have `capable-swine` in it, not checking if it was anywhere else in the output too ... 🐷 🐷 🐷

Turns out this swine wasn't so capable after all.

Reply:
:pig2: :see_no_evil: